src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/QueryInterfaceTests.Windows.cs

```diff
@@ -0,0 +1,90 @@
+            yield return new object[] { new NonDualComObjectEmpty(), IID_IDISPATCH };
+            yield return new object[] { new AutoDispatchComObjectEmpty(), IID_IUNKNOWN };
+            yield return new object[] { new AutoDispatchComObjectEmpty(), IID_IDISPATCH };
+            yield return new object[] { new AutoDualComObjectEmpty(), IID_IUNKNOWN };
+            yield return new object[] { new AutoDualComObjectEmpty(), IID_IDISPATCH };
+
+            yield return new object[] { new IUnknownComObject(), IID_IUNKNOWN };
+            yield return new object[] { new IUnknownComObject(), IID_IDISPATCH };
+        }
+    
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(QueryInterface_ValidComObjectInterface_TestData))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void QueryInterface_ValidComObjectInterface_Success(object o, string iidString)
+        {
+            QueryInterface_ValidInterface_Success(o, iidString);
+        }
+
+        public static IEnumerable<object[]> QueryInterface_NoSuchComObjectInterface_TestData()
+        {
+            yield return new object[] { new ComImportObject(), IID_IINSPECTABLE };
+            yield return new object[] { new ComImportObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+
+            yield return new object[] { new DualComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new DualComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+            yield return new object[] { new IUnknownComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new IUnknownComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+            yield return new object[] { new IDispatchComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new IDispatchComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+            yield return new object[] { new IInspectableComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new IInspectableComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+
+            yield return new object[] { new NonDualComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new NonDualComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+            yield return new object[] { new AutoDispatchComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new AutoDispatchComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+            yield return new object[] { new AutoDualComObject(), IID_IINSPECTABLE };
+            yield return new object[] { new AutoDualComObject(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
+
+            yield return new object[] { new NonDualComObjectEmpty(), IID_IINSPECTABLE };
+            yield return new object[] { new NonDualComObjectEmpty(), "927971f5-0939-11d1-8be1-00c04fd8d503" };
```

Comment:
nit: Could you create a variable for guid "927971f5-0939-11d1-8be1-00c04fd8d503" ?